app/api/csv/importEntity.ts

```diff
@@ -66,6 +71,120 @@ type Options = {
   language: string;
 };
 
+const filterJSObject = (input: { [k: string]: any }, keys: string[]): { [k: string]: any } => {
+  const result: { [k: string]: any } = {};
+  keys.forEach(k => {
+    if (input.hasOwnProperty(k)) {
+      result[k] = input[k];
+    }
+  });
+  return result;
+};
+
+const arrangeThesauri = async (
```

Comment:
@LaszloKecskes this function is really difficult to follow and understand, i have write a couple of comments, but i think its best if this is first refactored before proceeding with more code review (there is an issue on codeclimate pointing out this cognitive complexity).
some general things i think are important and can improve the complexity, readability:

- move this into a separate file, this clearly does not belong to importEntity as does another thing.
- try to maintain as less state as possible, there is 4 variables that you need to keep track when reading the function.
- split the function into smaller pieces with clear responsabilities
- throw errors instead of using the ```errorContext``` strategy (pointed out in another comment)

Reply:
Moved, and split up. As discussed, there's nothing I see about the state, I've tried to organize it more clearly.
It turns out the thesauri saving at the end was responsible for the most of it, as it's still a 7 over 5 on code climate. Can we pair that?

Reply:
sure lets do it !